src/components/Table/GridTable.stories.tsx

```diff
@@ -134,6 +134,66 @@ export function VirtualFiltering() {
   );
 }
 
+export function VirtualFilteringWithFilterablePin() {
+  const rows: GridDataRow<Row>[] = useMemo(
+    () => [
+      simpleHeader,
+      ...zeroTo(1_000).map((i) => ({
+        kind: "data" as const,
+        id: String(i),
+        data: {
+          name:
+            i === 200
```

Comment:
Thought: I know this is super picky, but math like this is something we try to avoid (see the "Generally Avoid Math in Tests" section https://homebound.atlassian.net/wiki/spaces/ENG/pages/2347171841/Testing+Style+Guide)

The story you're copy/pasting had 1,000 rows precisely b/c it wanted to test how "really large tables" behavior/visualize/interact; for what you're doing, we could/should just ~2-3 rows, make it as dumb/simple as possible, so that when we open up the story in ~6-9 months we know "okay, okay, I only have to think about these three rows" and not the other ~997 we're making that are just going to be a distraction.